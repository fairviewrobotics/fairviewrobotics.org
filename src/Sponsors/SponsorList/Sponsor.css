.company {
  margin: 3vw;
  opacity: 0.8;
  transition: opacity 0.25s;
}

.company:hover {
  opacity: 1;
}

@media(max-width:768px) {
  .company {
    margin: 15vw;
    transform: scale(1.8);
  }
}