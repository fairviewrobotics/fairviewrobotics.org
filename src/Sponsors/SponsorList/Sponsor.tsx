import React, { PureComponent } from 'react';

import styles from './Sponsor.module.css';

export type SponsorProps = {
  name: string;
  size: number;
  url: string;
  src: string;
}

export default class Sponsor extends PureComponent<SponsorProps> {
  render() {
    const trueWidth = this.props.size * .9;

    return (
      <a className={styles.sponsor} target='_blank' href={this.props.url} rel="noopener noreferrer">
        <img src={this.props.src} alt={this.props.name} className={styles.image} style={{ width: `${trueWidth}vw` }} />
      </a>
    );
  }
}
