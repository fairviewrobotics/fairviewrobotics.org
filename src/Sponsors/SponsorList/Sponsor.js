import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';

import './Sponsor.css';

export default class Sponsor extends PureComponent {

  static propTypes = {
    name: PropTypes.string.isRequired,
    size: PropTypes.number.isRequired,
    url: PropTypes.string.isRequired,
    src: PropTypes.string.isRequired
  };

  render() {
    const trueWidth = this.props.size * .9;

    return (
      <a className='company' target='_blank' href={this.props.url}>
        <img src={this.props.src} alt={this.props.name} style={{ height: 'auto', width: `${trueWidth}vw` }} />
      </a>
    );
  }
}
