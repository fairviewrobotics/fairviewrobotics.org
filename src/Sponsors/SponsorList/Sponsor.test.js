import React from "react";
import { shallow } from "enzyme";
import Sponsor from "./Sponsor";

describe('Sponsor', () => {
  let wrapper;

  it('renders without crashing', () => {
    shallow(<Sponsor src="" size={0} name="" url=""/>);
  });

  describe('given a sponsor', () => {
    beforeEach(() => {
      wrapper = shallow(
        <Sponsor name="Jason" size={50} url="urltosponsor.com" src="imageurlforsponsor"/>
      );
    });

    it('should render a clickable image', () => {
      expect(wrapper).toMatchSnapshot();
    });

    it('should have a size 90% of the view width', () => {
      // 50 * 90% = 45
      expect(wrapper.find('img')).toHaveStyle('width', '45vw');
    });
  });
});