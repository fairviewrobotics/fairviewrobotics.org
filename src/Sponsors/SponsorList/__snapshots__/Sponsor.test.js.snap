// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sponsor given a sponsor should render a clickable image 1`] = `
<a
  className="sponsor"
  href="urltosponsor.com"
  target="_blank"
>
  <img
    alt="Jason"
    className="image"
    src="imageurlforsponsor"
    style={
      Object {
        "width": "45vw",
      }
    }
  />
</a>
`;
